web_monitoring/db.py

```diff
@@ -1,47 +1,608 @@
+    ----------
+    annotation_id : string
+    page_id : string
+    to_version_id : string
+    from_version_id : string, optional
+        If from_version_id is not given, it will be treated as version
+        immediately prior to ``to_version``.
+
+    Returns
+    -------
+    response : dict
+    """
+    url = (f'{api_url()}/pages/{page_id}/changes/'
+           f'{from_version_id}..{to_version_id}/annotations/{annotation_id}')
+    res = requests.get(url, auth=auth())
+    res.raise_for_status()
+    result = res.json()
+    # In place, replace datetime strings with datetime objects.
+    data = result['data']
+    data['created_at'] = parse_timestamp(data['created_at'])
+    data['updated_at'] = parse_timestamp(data['updated_at'])
+    return result
+
+
+### CONVENIENCE FUNCTIONS ###
+
+
+@ensure_credentials
+def fetch_version_content(version_id):
+    """
+    Download the saved content from a given Version.
+
+    Parameters
+    ----------
+    version_id : string
+
+    Returns
+    -------
+    content : bytes
+    """
+    url = f'{api_url()}/versions/{version_id}'
+    res = requests.get(url, auth=auth())
+    assert res.ok
+    content_uri = res.json()['data']['uri']
+    return requests.get(content_uri).content
+
+
+def get_version_by_versionista_id(versionista_id):
+    """
+    Look up a Version by its Verisonista-issued ID.
+
+    This is a convenience function for dealing with Versions ingested from
+    Versionista.
+
+    Parameters
+    ----------
+    versionista_id : string
+
+    Returns
+    -------
+    response : dict
+    """
+    result = list_versions(source_type='versionista',
+                           source_metadata={'version_id': versionista_id})
+    # There should not be more than one match.
+    data = result['data']
+    if len(data) == 0:
+        raise ValueError("No match found for versionista_id {}"
+                         "".format(versionista_id))
+    elif len(data) > 1:
+        raise Exception("Multiple Versions match the versionista_id {}."
+                        "Their web-monitoring-db version_ids are: {}"
+                        "".format(versionista_id, [v['uuid'] for v in data]))
+    # Hack 'data' to look like the result of get_version rather than the result
+    # of list_versions.
+    result['data'] = result['data'][0]
+    return result
+
+
+def get_version_uri(version_id, id_type='db', source_type='versionista', get_previous=False):
+    """
+    Get the uri of a version(snapshot) stored in the db.
+
+    Parameters
+    ----------
+    version_id : string
+    source_type : string, optional
+        'versionista' by default
+    get_previous : boolean
+        False by default
+
+    Returns
+    -------
+    source_uri or (source_uri, previous_id) depending on ``get_previous``
+    """
+    if (id_type == 'db'):
+        result = get_version(version_id=version_id)
+    elif (id_type == 'source'):
+        result = get_version_by_versionista_id(versionista_id=version_id)
```

Comment:
“source” = “versionista” seems like a pretty brittle equivalence for the future.

Reply:
@janakrajchadha What motivated your choice of the word `'source'` as opposed to, say, `'versionista'` here?

Reply:
Then now might be a good time to remove this, as we are already aggressively breaking the API. It was a useful iteration for clarifying our thoughts about what we need. I expect your experiments will prompt more convenience methods in the future.